File: programmers/Level2/CPP/3진법 뒤집기.cpp
#include <string>
#include <vector>
#include <cmath>

using namespace std;

int solution(int n) {
    int answer = 0;
    
    string str = "";
    int i = 0;
    while(n != 0){
        str += n%3;
        n /= 3;
    }
    int j = 0;
    for(int i = str.size()-1; i>=0;i--){
        answer += str[i] * pow(3 , j);j++;
    }
    
    return answer;
}

File: programmers/Level1/JAVA/3진법 뒤집기.java
import java.util.*;

class Solution {
    public int solution(int n) {
       int answer = 0;
        String third = Integer.toString(n, 3);
        StringBuffer sb = new StringBuffer(third);
        String reversed = sb.reverse().toString();

        int exp = 0;
        for (int i = reversed.length() - 1; i >= 0; i--) {
            answer += Integer.parseInt(String.valueOf(reversed.charAt(i))) * Math.pow(3, exp);
            exp++;
        }

        return answer;
    
    }
}
